src/gui/closeabletabwidget.h

```diff
@@ -0,0 +1,47 @@
+/*
+ * Bittorrent Client using Qt and libtorrent.
+ * Copyright (C) 2020  Will Da Silva <will@willdasilva.xyz>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * In addition, as a special exception, the copyright holders give permission to
+ * link this program with the OpenSSL project's "OpenSSL" library (or with
+ * modified versions of it that use the same license as the "OpenSSL" library),
+ * and distribute the linked executables. You must obey the GNU General Public
+ * License in all respects for all of the code used other than "OpenSSL".  If you
+ * modify file(s), you may extend this exception to your version of the file(s),
+ * but you are not obligated to do so. If you do not wish to do so, delete this
+ * exception statement from your version.
+ */
+
+#ifndef CLOSEABLETABWIDGET_H
+#define CLOSEABLETABWIDGET_H
+
+#include <QTabBar>
+#include <QTabWidget>
+
+class MiddleClickTabBar: public QTabBar
+{
+public:
+    void mouseReleaseEvent(QMouseEvent* event) override;
+};
+
+class CloseableTabWidget final : public QTabWidget
```

Comment:
On a closer look, seems this class is redundant too.
At searchwidget we should able to replace the tabbar with `QTabWidget::setTabBar()`.

Reply:
`setTabBar` is protected.

Reply:
Why not just set an event filter on a regular tab widget (its tab bar) inside SearchWidget, instead of creating these wrappers? It's not like we need them to be reused anywhere else.
In any case, isn't this name confusing? Is the widget itself Closeable?

Reply:
>Why not just set an event filter on a regular tab widget (its tab bar) inside SearchWidget, instead of creating these wrappers? It's not like we need them to be reused anywhere else.

Agree.
@WillDaSilva Seems doable to me, would you explore this possibility? https://doc.qt.io/qt-5/eventsandfilters.html#event-filters

>In any case, isn't this name confusing? Is the widget itself Closeable?

Let's find a better name, it is hard for me to think of something other than `CustomTabWidget`...

Reply:
> Let's find a better name, it is hard for me to think of something other than `CustomTabWidget`...

Perhaps it is really difficult to find a suitable name here. But even `CustomTabWidget` is better.
I would still try event filtering first to avoid create such class.